operator/pkg/cli/migrate_v2_pod.go

```diff
@@ -0,0 +1,339 @@
+	apitypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	migrateV2PodNamespace = "kotsadm"
+	migrateV2PodName      = "migrate-v2"
+)
+
+var _migrateV2PodSpec = corev1.Pod{
+	TypeMeta: metav1.TypeMeta{
+		APIVersion: "v1",
+		Kind:       "Pod",
+	},
+	ObjectMeta: metav1.ObjectMeta{
+		Namespace: migrateV2PodNamespace,
+		Name:      migrateV2PodName,
+		Labels: map[string]string{
+			"app": "install-v2-manager",
+		},
+	},
+	Spec: corev1.PodSpec{
+		ServiceAccountName: "kotsadm",
+		RestartPolicy:      corev1.RestartPolicyNever,
+		Containers: []corev1.Container{
+			{
+				Name:            "install-v2-manager",
+				Image:           "DYNAMIC",
+				ImagePullPolicy: corev1.PullIfNotPresent,
+				Command: []string{
+					"/manager", "migrate-v2",
+					"--installation", "/ec/installation/installation",
+					// "--migrate-v2-secret", "DYNAMIC",
+					// "--app-slug", "DYNAMIC",
+					// "--app-version-label", "DYNAMIC",
+				},
+				VolumeMounts: []corev1.VolumeMount{
+					{
+						Name:      "installation", // required to set runtime config
+						MountPath: "/ec/installation",
+						ReadOnly:  true,
+					},
+				},
+			},
+		},
+		Volumes: []corev1.Volume{
+			// {
+			// 	Name: "installation",
+			// 	VolumeSource: corev1.VolumeSource{
+			// 		ConfigMap: &corev1.ConfigMapVolumeSource{
+			// 			LocalObjectReference: corev1.LocalObjectReference{
+			// 				Name: "DYNAMIC",
+			// 			},
+			// 		},
+			// 	},
+			// },
+		},
+	},
+}
+
+// LogFunc can be used as an argument to Run to log messages.
+type LogFunc func(string, ...any)
+
+// runMigrateV2PodAndWait runs the v2 migration pod and waits for the pod to finish.
+func runMigrateV2PodAndWait(
+	ctx context.Context, logf LogFunc, cli client.Client,
+	in *ecv1beta1.Installation,
+	migrationSecret string, appSlug string, appVersionLabel string,
+) error {
+	logf("Ensuring installation config map")
+	if err := ensureInstallationConfigMap(ctx, cli, in); err != nil {
+		return fmt.Errorf("ensure installation config map: %w", err)
+	}
+	logf("Successfully ensured installation config map")
+
+	logf("Getting operator image name")
+	operatorImage, err := getOperatorImageName(ctx, cli, in)
+	if err != nil {
+		return fmt.Errorf("get operator image name: %w", err)
+	}
+	logf("Successfully got operator image name")
+
+	logf("Ensuring v2 migration pod")
+	_, err = ensureMigrateV2Pod(ctx, cli, in, operatorImage, migrationSecret, appSlug, appVersionLabel)
+	if err != nil {
+		return fmt.Errorf("create pod: %w", err)
+	}
+	logf("Successfully ensured v2 migration pod")
+
+	logf("Waiting for v2 migration pod to finish")
+	err = waitForMigrateV2Pod(ctx, cli)
+	if err != nil {
+		return fmt.Errorf("wait for pod: %w", err)
+	}
+	logf("Successfully waited for v2 migration pod to finish")
+
+	// NOTE: the installation config map cannot be deleted because the service account gets deleted
```

Comment:
why would you even want to delete the configmap?

Reply:
just to clean things up